packages/web-components/src/checkbox/checkbox.ts

```diff
@@ -42,77 +158,208 @@ export class Checkbox extends FormAssociatedCheckbox {
+   */
+  private _value: string = this.initialValue;
+
+  /**
+   * The current value of the input.
+   * @public
+   */
+  public get value(): string {
+    Observable.track(this, 'value');
+    return this._value;
+  }
+
+  public set value(value: string) {
+    this._value = value;
+
+    if (this.$fastController.isConnected) {
+      this.setFormValue(value);
+      this.setValidity();
+      Observable.notify(this, 'value');
+    }
+  }
+
+  /**
+   * The associated form element.
    *
    * @public
-   * @default 'after'
    * @remarks
-   * HTML Attribute: label-position
+   * Reflects the {@link https://developer.mozilla.org/docs/Web/API/ElementInternals/form | `ElementInternals.form`} property.
    */
-  @attr({ attribute: 'label-position' })
-  public labelPosition?: CheckboxLabelPosition;
+  public get form(): HTMLFormElement | null {
+    return this.elementInternals.form;
+  }
+
   /**
-   * The element's value to be included in form submission when checked.
-   * Default to "on" to reach parity with input[type="checkbox"]
+   * The internal {@link https://developer.mozilla.org/docs/Web/API/ElementInternals | `ElementInternals`} instance for the component.
    *
    * @internal
    */
-  public initialValue: string = 'on';
+  protected elementInternals: ElementInternals = this.attachInternals();
 
   /**
-   * @internal
+   * The form-associated flag.
+   * @see {@link https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-face-example | Form-associated custom elements}
+   *
+   * @public
    */
-  @observable
-  public defaultSlottedNodes!: Node[];
+  public static formAssociated = true;
 
   /**
    * The indeterminate state of the control
    */
   @observable
   public indeterminate: boolean = false;
 
+  /**
+   * Synchronizes the element's indeterminate state with the internal ElementInternals state.
+   *
+   * @internal
+   */
+  public indeterminateChanged(prev: boolean, next: boolean): void {
+    this.elementInternals.ariaChecked = next ? 'mixed' : `${this.checked}`;
+
+    if (next) {
+      this.setAttribute('aria-checked', 'mixed');
+      return;
+    }
+
+    if (this.checked) {
+      this.setAttribute('aria-checked', 'true');
+      return;
+    }
+
+    this.removeAttribute('aria-checked');
+  }
+
+  /**
+   * The element's validity state.
+   *
+   * @public
+   * @remarks
+   * Reflects the {@link https://developer.mozilla.org/docs/Web/API/ElementInternals/validity | `ElementInternals.validity`} property.
+   */
+  public get validity(): ValidityState {
+    return this.elementInternals.validity;
+  }
+
+  public connectedCallback() {
+    super.connectedCallback();
+
+    this.setFormValue(this.checked ? this.value : null);
+    this.setValidity();
+    this.elementInternals.ariaChecked = `${this.checked}`;
+
+    this.addEventListener('click', this.clickHandler);
+    this.addEventListener('keypress', this.keydownHandler);
+    this.addEventListener('input', this.inputHandler);
```

Comment:
Can we bind these in the template to just make use of the bindings and keep things consistent?